lib/Driver/Driver.cpp

```diff
@@ -1700,15 +1746,35 @@ void Driver::buildJobs(ArrayRef<const Action *> TopLevelActions,
 
   for (const Action *A : TopLevelActions) {
     if (auto *JA = dyn_cast<JobAction>(A)) {
-      (void)buildJobsForAction(C, JA, OI, OFM, TC, /*TopLevel*/true, JobCache);
+      (void)buildJobsForAction(C, JA, OI, OFM, workingDirectory, TC,
+                               /*TopLevel=*/true, JobCache);
     }
   }
 }
 
+/// Form a filename based on \p base in \p result, optionally setting its
+/// extension to \p newExt and in \p workingDirectory.
+static void filenameFromBaseAndExt(StringRef base, StringRef newExt,
```

Comment:
Nitpick: I think this function name needs a verb if the result isn't in the return value.